tests/testthat/test-assignment_linter.R

```diff
@@ -21,3 +21,14 @@ test_that("returns the correct linting", {
     linter
   )
 })
+
+test_that("other operators are caught as well", {
+  expect_lint("1 -> blah", rex::rex("Use <-, not ->, for assignment."), assignment_linter())
+
+  # <<- and ->> are only blocked optionally
+  expect_lint("1 <<- blah", NULL, assignment_linter())
+  expect_lint("1 ->> blah", NULL, assignment_linter())
```

Comment:
Seems odd to disallow `->` and allow `->>`.
Maybe better to change the configuration to

`allow_parent_assign = TRUE, allow_right_arrow = FALSE`

WDYT?